### ISSUE TITLE
reproject_gcps results in erroneous geolocation

### ISSUE THREAD

**Issue by mortenwh** [nansencenter/nansat#376]
```
In [1]: from nansat.nansat import Nansat
In [2]: n0 = Nansat('/mnt/10.11.12.232/sat_downloads/quikscat/l2b12/v3/2008/049/qs_l2b_45128_v3_200802180005.nc', quartile=0)
In [3]: n0.get_border()
Out[3]: 
(array([ -21.28056745,  -20.41340618,  -18.87628676,  -16.66920919,
         -13.79217348,  -10.24517962,   -6.02822761,   -1.14131745,
           4.41555086,   10.64237732,   18.50935226,   18.50935226,
         -29.23742757,  -68.7482625 , -100.02315253, -123.06209766,
        -137.86509788, -144.43215321, -142.76326364, -132.85842916,
        -114.71764979,  -87.58545946,  -87.58545946,  -85.73871133,
         -83.39460567,  -80.38054186,  -76.6965199 ,  -72.34253979,
         -67.31860154,  -61.62470513,  -55.26085058,  -48.22703788,
         -40.52326704,  -40.52326704,  -76.53319789, -103.33779633,
        -121.90644987, -132.23915852, -134.33592226, -128.1967411 ,
        -113.82161504,  -91.21054408,  -60.36352821,  -21.28056745]),
 array([-88.59699188, -87.76478794, -86.88037144, -85.94374238,
        -84.95490075, -83.91384657, -82.82057982, -81.67510051,
        -80.47740863, -79.2275042 , -77.74782665, -77.74782665,
        -71.00631119, -64.01152084, -56.76345558, -49.26211542,
        -41.50750035, -33.49961039, -25.23844552, -16.72400575,
         -7.95629108,   1.29064303,   1.29064303,   1.03205803,
          0.85958819,   0.73933092,   0.6712862 ,   0.65545405,
          0.69183447,   0.78042744,   0.92123298,   1.11425107,
          1.35948174,   1.35948174,  -9.00345524, -18.86028114,
        -28.46383214, -37.81410824, -46.91110943, -55.75483572,
        -64.34528711, -72.6824636 , -80.76636519, -88.59699188]))
In [4]: n0.reproject_gcps()

In [5]: n0.get_border()
Out[5]: 
(array([  -85674.45985479,  -241415.3712833 ,  -399191.03869212,
         -559001.46208126,  -720846.64145071,  -884726.57680047,
        -1050641.26813055, -1218590.71544094, -1388574.91873164,
        -1560593.87800265, -1758008.492406  , -1758008.492406  ,
        -2525456.23565694, -3221656.21558208, -3846608.43218143,
        -4400312.88545499, -4882769.57540275, -5293978.50202472,
        -5633939.66532089, -5902653.06529127, -6100118.70193585,
        -6228551.74536919, -6228551.74536919, -5963651.53466752,
        -5732086.46101564, -5502556.14334407, -5275060.58165281,
        -5049599.77594187, -4826173.72621124, -4604782.43246092,
        -4385425.89469091, -4168104.11290122, -3952817.08709184,
        -3952817.08709184, -3886061.69382508, -3748787.49890934,
        -3540265.5406678 , -3260495.81910047, -2909478.33420734,
        -2487213.08598842, -1993700.0744437 , -1428939.29957319,
         -792930.76137689,   -85674.45985479]),
 array([-5429951.33041346, -5572755.52903473, -5716706.76929382,
        -5861805.05119073, -6008050.37472545, -6155442.73989798,
        -6303982.14670833, -6453668.5951565 , -6604502.08524249,
        -6756482.61696628, -6930113.86547764, -6930113.86547764,
        -5962937.98690378, -4963704.91333859, -3932414.64478206,
        -2869067.18123419, -1773662.52269498,  -646200.66916444,
          513318.37935745,  1704894.62287067,  2928528.06137522,
         4215628.94175584,  4215628.94175584,  4253098.66125118,
         4284936.66737262,  4315627.63185624,  4345171.55470204,
         4373568.43591003,  4400818.27548021,  4426921.07341256,
         4451876.82970711,  4475685.54436383,  4498347.21738274,
         4498347.21738274,  3335689.53100865,  2233500.65532973,
         1163368.97464215,   125294.48894591,  -880722.801759  ,
        -1854682.89747257, -2796585.7981948 , -3706431.50392569,
        -4584220.01466524, -5429951.33041346]))
```

**Comment by akorosov** [comment]
Which branch should I use to pen the file?
In master and in hotfix365... it gives an error:
``` python
In [2]: n = Nansat('/Data/sat/downloads/quikscat/l2b12/v3/2008/049/qs_l2b_45128_v3_200802180005.nc')
=>QuikSCAT Level 2B Ocean Wind Vectors in 12.5km Slice Composites<=
Warning 1: dimension #1 (cross_track) is not a Longitude/X dimension.
Warning 1: dimension #0 (along_track) is not a Latitude/Y dimension.
Warning 1: dimension #1 (cross_track) is not a Longitude/X dimension.
Warning 1: dimension #0 (along_track) is not a Latitude/Y dimension.
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
~/py/nansat/nansat/mappers/mapper_netcdf_cf.py in _band_dict(self, subfilename, band_num, subds, band, band_metadata)
    267                 band_metadata['time_iso_8601'] = self._time_count_to_np_datetime64(
--> 268                     band_metadata[timecountname])
    269             except KeyError as e:

KeyError: 'NETCDF_DIM_time'

During handling of the above exception, another exception occurred:

AttributeError                            Traceback (most recent call last)
<ipython-input-2-569c936f8f17> in <module>()
----> 1 n = Nansat('/Data/sat/downloads/quikscat/l2b12/v3/2008/049/qs_l2b_45128_v3_200802180005.nc')

~/py/nansat/nansat/nansat.py in __init__(self, filename, mapper, log_level, **kwargs)
    151         self._init_empty(filename, log_level)
    152         # Create VRT object with mapping of variables
--> 153         self.vrt = self._get_mapper(mapper, **kwargs)
    154
    155     def __getitem__(self, band_id):

~/py/nansat/nansat/nansat.py in _get_mapper(self, mappername, **kwargs)
   1140                 # create a Mapper object and get VRT dataset from it
   1141                 try:
-> 1142                     tmp_vrt = nansatMappers[iMapper](self.filename, gdal_dataset, metadata, **kwargs)
   1143                     self.logger.info('Mapper %s - success!' % iMapper)
   1144                     self.mapper = iMapper.replace('mapper_', '')

~/py/nansat/nansat/mappers/mapper_quikscat.py in __init__(self, filename, gdal_dataset, metadata, quartile, *args, **kwargs)
     24             raise WrongMapperError
     25
---> 26         super(Mapper, self).__init__(filename, gdal_dataset, metadata, quartile=quartile, *args, **kwargs)
     27
     28         band_lat = self.dataset.GetRasterBand(self._latitude_band_number(gdal_dataset))

~/py/nansat/nansat/mappers/scatterometers.py in __init__(self, filename, gdal_dataset, metadata, quartile, *args, **kwargs)
     18     def __init__(self, filename, gdal_dataset, metadata, quartile=0, *args, **kwargs):
     19
---> 20         super(Mapper, self).__init__(filename, gdal_dataset, metadata, *args, **kwargs)
     21
     22         intervals = [0,1,2,3]

~/py/nansat/nansat/mappers/mapper_netcdf_cf.py in __init__(self, filename, gdal_dataset, gdal_metadata, *args, **kwargs)
     64
     65         # Add bands with metadata and corresponding values to the empty VRT
---> 66         self.create_bands(self._band_list(gdal_dataset, metadata, *args, **kwargs))
     67
     68         # Check size?

~/py/nansat/nansat/mappers/mapper_netcdf_cf.py in _band_list(self, gdal_dataset, gdal_metadata, netcdf_dim, bands, *args, **kwargs)
    225                 # append band with src and dst dictionaries
    226                 bdict = self._band_dict(fn, band_num, subds, band=band,
--> 227                         band_metadata=band_metadata)
    228                 if bdict:
    229                     metadictlist.append(bdict)

~/py/nansat/nansat/mappers/mapper_netcdf_cf.py in _band_dict(self, subfilename, band_num, subds, band, band_metadata)
    273                 warnings.warn(
    274                         '%s: %s - %s Continuing without time metadata for band %s'
--> 275                         %(e.__repr__().split('(')[0], e.message, e.__doc__,
    276                             band_metadata['NETCDF_VARNAME']))
    277

AttributeError: 'KeyError' object has no attribute 'message'

```

**Comment by akorosov** [comment]
The errror above also comes if I use the branch hotfix365 and add `quartile=0`. In the master branch, if I specify quartile it gives another error:
```
In [2]: n = Nansat('/Data/sat/downloads/quikscat/l2b12/v3/2008/049/qs_l2b_45128_v3_200802180005.nc', quartile=0)
=>QuikSCAT Level 2B Ocean Wind Vectors in 12.5km Slice Composites<=
Warning 1: dimension #1 (cross_track) is not a Longitude/X dimension.
Warning 1: dimension #0 (along_track) is not a Latitude/Y dimension.
Warning 1: dimension #1 (cross_track) is not a Longitude/X dimension.
Warning 1: dimension #0 (along_track) is not a Latitude/Y dimension.
Warning 1: dimension #1 (cross_track) is not a Longitude/X dimension.
Warning 1: dimension #0 (along_track) is not a Latitude/Y dimension.
Warning 1: dimension #1 (cross_track) is not a Longitude/X dimension.
Warning 1: dimension #0 (along_track) is not a Latitude/Y dimension.
Warning 1: dimension #1 (cross_track) is not a Longitude/X dimension.
Warning 1: dimension #0 (along_track) is not a Latitude/Y dimension.
Warning 1: dimension #1 (cross_track) is not a Longitude/X dimension.
Warning 1: dimension #0 (along_track) is not a Latitude/Y dimension.
Warning 1: dimension #1 (cross_track) is not a Longitude/X dimension.
Warning 1: dimension #0 (along_track) is not a Latitude/Y dimension.
Warning 1: dimension #1 (cross_track) is not a Longitude/X dimension.
Warning 1: dimension #0 (along_track) is not a Latitude/Y dimension.
Warning 1: dimension #1 (cross_track) is not a Longitude/X dimension.
Warning 1: dimension #0 (along_track) is not a Latitude/Y dimension.
Warning 1: dimension #1 (cross_track) is not a Longitude/X dimension.
Warning 1: dimension #0 (along_track) is not a Latitude/Y dimension.
Warning 1: dimension #1 (cross_track) is not a Longitude/X dimension.
Warning 1: dimension #0 (along_track) is not a Latitude/Y dimension.
Warning 1: dimension #1 (cross_track) is not a Longitude/X dimension.
Warning 1: dimension #0 (along_track) is not a Latitude/Y dimension.
Warning 1: dimension #1 (cross_track) is not a Longitude/X dimension.
Warning 1: dimension #0 (along_track) is not a Latitude/Y dimension.
/home/antonk/py/nansat/nansat/mappers/mapper_netcdf_cf.py:335: UserWarning: GDAL cannot determine the dataset projection - using Nansat spatial reference WKT, assuming a regular longitude/latitude grid
  warnings.warn('GDAL cannot determine the dataset projection - using Nansat ' \
Warning 1: dimension #1 (cross_track) is not a Longitude/X dimension.
Warning 1: dimension #0 (along_track) is not a Latitude/Y dimension.
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-2c7b636a0a7f> in <module>()
----> 1 n = Nansat('/Data/sat/downloads/quikscat/l2b12/v3/2008/049/qs_l2b_45128_v3_200802180005.nc', quartile=0)

~/py/nansat/nansat/nansat.py in __init__(self, filename, mapper, log_level, **kwargs)
    151         self._init_empty(filename, log_level)
    152         # Create VRT object with mapping of variables
--> 153         self.vrt = self._get_mapper(mapper, **kwargs)
    154
    155     def __getitem__(self, band_id):

~/py/nansat/nansat/nansat.py in _get_mapper(self, mappername, **kwargs)
   1139                 # create a Mapper object and get VRT dataset from it
   1140                 try:
-> 1141                     tmp_vrt = nansatMappers[iMapper](self.filename, gdal_dataset, metadata, **kwargs)
   1142                     self.logger.info('Mapper %s - success!' % iMapper)
   1143                     self.mapper = iMapper.replace('mapper_', '')

~/py/nansat/nansat/mappers/mapper_netcdf_cf.py in __init__(self, filename, gdal_dataset, gdal_metadata, *args, **kwargs)
     64
     65         # Add bands with metadata and corresponding values to the empty VRT
---> 66         self.create_bands(self._band_list(gdal_dataset, metadata, *args, **kwargs))
     67
     68         # Check size?

TypeError: _band_list() got an unexpected keyword argument 'quartile'
```

**Comment by akorosov** [comment]
Also, I checked with the tests/data/gcps.tif file and there get_bored() works fine.

**Comment by mortenwh** [comment]
I must have used hotfix365 branch - sorry for being unclear. Seems the KeyError issue is the python 3 issue that you commented about in the pull request, right? Could you try to test with python 2.7 unless we first solve the python 3 issue?

**Comment by mortenwh** [comment]
Testing again now in version 27b80e13b5aeca50605dcc4ff79496d91acefc4a and Python 2.7, I still get the same error for the file above (`/mnt/10.11.12.232/sat_downloads/quikscat/l2b12/v3/2008/049/qs_l2b_45128_v3_200802180005.nc`). The same procedure with `tests/data/gcps.tif` works as expected...

**Comment by mortenwh** [comment]
The key error should now be solved in #365

**Comment by mortenwh** [comment]
@akorosov - could you check again?

**Comment by mortenwh** [comment]
Solved in eff81a36e0e0dbeb67cff9b9047bc08ae4cab587 for #365
